File: libs/jetpack/lib-arch/src/main/java/com/yyxnb/what/arch/annotations/SwipeStyle.java
package com.yyxnb.what.arch.annotations;


import androidx.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * 侧滑 注解
 *
 * @author yyx
 */
@IntDef({SwipeStyle.FULL, SwipeStyle.EDGE, SwipeStyle.NONE})
@Retention(RetentionPolicy.SOURCE)
public @interface SwipeStyle {
    /**
     * 全屏
     */
    int FULL = 1;
    /**
     * 边缘
     */
    int EDGE = 2;
    /**
     * none
     */
    int NONE = 0;
}

File: widget/lib_arch/src/main/java/com/yyxnb/arch/annotations/SwipeStyle.kt
package com.yyxnb.arch.annotations

import androidx.annotation.IntDef

/**
 * 侧滑 注解
 *
 * @author yyx
 */
@IntDef(SwipeStyle.FULL, SwipeStyle.EDGE, SwipeStyle.NONE)
@kotlin.annotation.Retention(AnnotationRetention.SOURCE)
annotation class SwipeStyle {
    companion object {
        /**
         * 全屏
         */
        const val FULL = 1

        /**
         * 边缘
         */
        const val EDGE = 2

        /**
         * none
         */
        const val NONE = 0
    }
}